nilearn/input_data/base_masker.py

```diff
@@ -97,6 +97,11 @@ def filter_and_extract(imgs, extraction_function,
                                 func_memory_level=2,
                                 memory_level=memory_level)(imgs)
 
+    if np.any(np.isnan(region_signals)):
+        ensure_finite = True
+    else:
+        ensure_finite = False
```

Comment:
Actually, I've just realized that you are doing the ensure_finite on the signals, while it should really be done on the maps. I'll change that.